launch/org.eclipse.cdt.docker.launcher/src/org/eclipse/cdt/internal/docker/launcher/ContainerLaunchUtils.java

```diff
@@ -82,4 +92,52 @@ public static final String toDockerVolume(Map<String, String> pMap, IPath path)
 		return rv;
 	}
 
+	/**
+	 * Pull an Docker-Image if is not locally available.
+	 *
+	 * The interface is rather bad, as it returns null on success to keep things simple at the call site.
+	 * This should probably be moved to linuxtools anyway.
+	 *
+	 * @param monitor A Monitor to show progress
+	 * @param connectionName The name of the connection
+	 * @param imageName The image to pull
+	 * @return null on success, otherwise an error string
+	 *
+	 */
+	public static String provideDockerImage(IProgressMonitor monitor, @NonNull final String connectionName,
+			@NonNull final String imageName) {
+
+		// Try to pull image, if necessary
+		final var connection = (IDockerConnection4) DockerConnectionManager.getInstance()
+				.getConnectionByName(connectionName);
+
+		if (connection == null) {
+			// This is unlikely to happen, but who knows
+			return NLS.bind(Messages.ContainerCommandLauncher_pullerr_noConn, imageName, connectionName);
+		}
+
+		// See if the image already exists
+		if (connection.getImageInfo(imageName) != null)
+			return null;
+
+		try {
+			final var prghandler = connection.getDefaultPullImageProgressHandler(imageName, monitor);
+			connection.pullImage(imageName, prghandler);
+			return null;
+		} catch (DockerException e) {
+			final String rv;
```

Comment:
I think naming this errMsg might make the code easier to understand.

Reply:
As I'm returning an IStatus now, I stuck with `rv`.